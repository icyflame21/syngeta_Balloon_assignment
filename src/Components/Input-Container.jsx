import React, { useState} from "react";

export const Input_Container = () => {
    const [value, setValue] = useState("");
    
// onSubmit function udpates the state "setValue" as per the entered the number , afterwards respective styles is been applied to each circle present in the circle container
    
  const onSubmit = () => {
    if (value > 5 || value < 1) alert("Please enter a number between 1 to 5");
    let val = document.getElementById(`empty-circle-${value}`);
    if (val.style.display === "block")
      alert(
        "Number already present in the container\nEnter a different number"
      );

    document.getElementById(`circle-${value}`).style.display = "none";
    document.getElementById(`empty-circle-${value}`).style.display = "block";
    setValue("");
  };
  return (
    <div className="input-container">
      <input
        type="number"
        id="number"
        autoFocus={true}
        placeholder="Enter a number between 1 to 5"
        value={value}
        onChange={(e) => {
          setValue(e.target.value);
        }}
      />
      <button className="submit" onClick={onSubmit}>
        Shoot
      </button>
    </div>
  );
};
