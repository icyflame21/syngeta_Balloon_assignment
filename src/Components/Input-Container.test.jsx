import { shallow } from 'enzyme';
import React from 'react'
import { Input_Container } from "./Input-Container";


it('check is Input_Container is loaded', () => {
  const c = shallow(<Input_Container />);
  expect(c).toBeTruthy()
});

it('test to match snapshot', () => {
    const component = shallow(<Input_Container />);
    expect(component.getElements()).toMatchSnapshot();
});