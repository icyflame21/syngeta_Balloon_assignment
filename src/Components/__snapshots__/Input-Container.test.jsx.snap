// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to match snapshot 1`] = `
Array [
  <div
    className="input-container"
>
    <input
        autoFocus="true"
        id="number"
        onChange={[Function]}
        placeholder="Enter a number between 1 to 5"
        type="number"
        value=""
    />
    <button
        className="submit"
        onClick={[Function]}
    >
        Shoot
    </button>
</div>,
]
`;
